.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,59 @@
+name: Benchmark
+
+on:
+- pull_request
+
+jobs:
+  benchmark:
+    strategy:
+      matrix:
+        ref:
+        - master
+        - ${{ github.sha }}
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Steup Go
+      uses: actions/setup-go@v2
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ matrix.ref }}
+
+    - name: Run Benchmarks
+      run: go test -v -run '^$$' -bench '(Marshal|Unmarshal)$$/codeResponse' -benchmem -cpu 1 -count 5 ./json | tee bench.txt
+
+    - name: Upload Benchmarks
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ matrix.ref }}
+        path: bench.txt
+
+  benchstat:
+    needs: [benchmark]
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Steup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17
+
+    - name: Setup Benchstat
+      run: go install golang.org/x/perf/cmd/benchstat@latest
```

Comment:
@achille-roussel I'm curious to see in practice how consistent results are when running benchmarks on multi-tenant GitHub Actions runners. I've seen it vary quite wildly.

Reply:
Yeah, it looks like the results are a bit random, probably depends on the VM that the task lands on. This is mostly interesting to showcase how to automate this workflow.

I think we'll need to use self-hosted runners on hosts that are designed to run benchmarks of this kind if we wanted to get more meaningful results.